src/shared-components/accordion/style.ts

```diff
@@ -22,29 +22,35 @@ export const Content = styled.div`
   width: 100%;
 `;
 
-export const ExpansionWrapper = styled.div`
-  max-height: ${({ contentHeight }) => contentHeight};
+export const ExpansionWrapper = styled.div<{ contentHeight: string }>`
+  max-height: ${({ contentHeight }): string => contentHeight};
   overflow: hidden;
   transition: max-height ${ANIMATION.defaultTiming} ease-in-out;
 `;
 
-const getBorderStyle = isOpen => css`
+const getBorderStyle = (isOpen: boolean): SerializedStyles => css`
   border: ${border};
 
   ${ExpansionWrapper} {
     ${isOpen && `border-top: ${border};`};
   }
 `;
 
-export const AccordionBox = styled.div`
-  ${({ noBorder, isOpen }) => !noBorder && getBorderStyle(isOpen)};
+export const AccordionBox = styled.div<{
```

Comment:
[Nit] I think as much as possible we should try to be consistent with the values we provide in the tagged templates. I think it is undesirable to have a boolean `false` returned to what is, effectively, supposed to be css markup. I've revised this function like so:

```
export const AccordionBox = styled.div<{
  noBorder: boolean;
  isOpen: boolean;
  disabled: boolean;
}>`
  ${({ noBorder, isOpen }): SerializedStyles | string =>
    !noBorder ? getBorderStyle(isOpen) : ``};

  width: 100%;

  &:not(:last-child) {
    border-bottom: none;
  }

  ${({ disabled }): string =>
    disabled
      ? `
    opacity: 0.4;
    background-color: ${COLORS.disabled};
    border-color: ${COLORS.purple30};
  `
      : ``};
`;
```

Reply:
done